src/test/java/com/shapesecurity/shift/validator/ValidatorTest.java

```diff
@@ -0,0 +1,536 @@
+/*
+ * Copyright 2014 Shape Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.shapesecurity.shift.validator;
+
+import com.shapesecurity.functional.data.ImmutableList;
+import com.shapesecurity.functional.data.Maybe;
+import com.shapesecurity.shift.ast.*;
+import com.shapesecurity.shift.parser.JsError;
+import com.shapesecurity.shift.parser.Parser;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ValidatorTest {
+
+  private void assertCorrectFailures(Script script, int expectedNumErrors, String expectedErrorMsg) {
+    ImmutableList<ValidationError> errors = Validator.validate(script);
+    assertEquals(expectedNumErrors, errors.length);
+    for (ValidationError error : errors) {
+      assertEquals(error.message, expectedErrorMsg);
+    }
+  }
+
+  private void assertCorrectFailures(Module module, int expectedNumErrors, String expectedErrorMsg) {
+    ImmutableList<ValidationError> errors = Validator.validate(module);
+    assertEquals(expectedNumErrors, errors.length);
+    for (ValidationError error : errors) {
+      assertEquals(error.message, expectedErrorMsg);
+    }
+  }
+
+  @Test
+  public void testBindingIdentifier() throws JsError {
+    Script test0 = Parser.parseScript("x=1");
+    assertCorrectFailures(test0, 0, "");
```

Comment:
I don't like this. Let's just have an `assertNoErrors` helper that only takes the AST.